source/pod.go

```diff
@@ -133,14 +142,18 @@ func (ps *podSource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, error
 			}
 
 			if domainAnnotation, ok := pod.Annotations[kopsDNSControllerHostnameAnnotationKey]; ok {
-				domainList := splitHostnameAnnotation(domainAnnotation)
-				for _, domain := range domainList {
-					node, _ := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
-					for _, address := range node.Status.Addresses {
-						recordType := suitableType(address.Address)
-						// IPv6 addresses are labeled as NodeInternalIP despite being usable externally as well.
-						if address.Type == corev1.NodeExternalIP || (address.Type == corev1.NodeInternalIP && recordType == endpoint.RecordTypeAAAA) {
-							addToEndpointMap(endpointMap, domain, recordType, address.Address)
+				if ps.nodeInformer == nil {
+					log.Debugf("Unable to determine node's hostname for pod %s/%s as node informer is disabled", pod.Namespace, pod.Name)
+				} else {
+					domainList := splitHostnameAnnotation(domainAnnotation)
+					for _, domain := range domainList {
+						node, _ := ps.nodeInformer.Lister().Get(pod.Spec.NodeName)
```

Comment:
What if the node not exist by now? Shell we handle error here? Could be a potential panic or CrashLoop for no reason